activerecord/lib/active_record/enum.rb

```diff
@@ -198,6 +198,11 @@ def _enum(name, values, prefix: nil, suffix: nil, scopes: true, **options)
 
         value_method_names = []
         _enum_methods_module.module_eval do
+          define_method("#{name}=") do |value|
+            raise UnknownAttributeError.new(self, name) unless self.class.columns_hash.key?(name)
+            super(value)
+          end
```

Comment:
Hum, I really don't like this overhead :/

The core of the problem is that we have to define the enum when the model is loaded, but it's only later, when the schema is loaded, that we'll know if the column exist.

Having an overhead on every assignment doesn't feel right when this could be checked only once when the schema is loaded.

I think there are some precedent for module that redefine the `load_schema` method (not sure what the actual name is just like that) to use it as a hook to validate some previous declarations.

Reply:
Ah it's `load_schema!`, and here's the example I had in mind: https://github.com/rails/rails/blob/3661d0d8a46a2725b3e7bacbc230e4f183fb67dc/activerecord/lib/active_record/attributes.rb#L263-L269


It's still not quite ideal because the error would be raised much later than when the error is made, but still better than on the first assignment. Especially in prod where the first assignment will be during a request, whereas loading the schema is at boot, so you fail the deploy vs serving 500s.

Reply:
> Especially in prod where the first assignment will be during a request

I hope the first assignment is in people's tests, not in prod 😄 

@byroot  Thank you for the review, as usual ❤️ Changed with the suggestion.